packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -196,9 +198,134 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         });
     }
 
+    /**
+     * Returns the root folder URI that a file belongs to.
+     * In the case that a file belongs to more than one root folders, returns the root folder closest to the file.
+     * If the file is not from the current workspace, returns an empty URI.
+     * @param {string} filePath - path of the file.
+     * @param {stirng[]} rootUris - URIs of the root folders in the workspace
+     * @returns URI of the root folder.
+     */
+    getRoot(filePath: string, rootUris: string[]): URI {
+        const roots = rootUris.filter(root => new URI(root).withScheme('file').isEqualOrParent(FileUri.create(filePath).withScheme('file')));
```

Comment:
The `FileUri` usage is problematic here, you may have to try using a different API, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#uripath

Reply:
I will change the part related to URI. Thanks for pointing it out!